rse-bash.Rmd

```diff
@@ -1,906 +1,627 @@
-This can be intimidating since it requires the knowledge of a few commands to get started,
-but just as when learning a natural language you can give quickly give
-efficient, precise instructions as your proficiency increases.
-Command-line interfaces are particularly good at automating repetitive tasks.
-For example,
-suppose we needed to copy the third line of each of a thousand text files stored
-in thousand different directories and paste it into a single file line by line.
-Using the traditional GUI approach of mouse clicks,
-this task would take several hours.
-With a command-line interface,
-a single instruction can be repeated (as is or with some modification) as many times as we want,
-so this task could be accomplished in a few minutes at most.
-
-The heart of a command-line interface is a **read-evaluate-print loop** (REPL). It is called
-so because when you type a command and press <kbd>Return</kbd>
-(also known as <kbd>Enter</kbd>) the command-line interface
-reads your command,
-evaluates (or "executes") it,
-prints the output of your command,
-then loops back and waits for you to enter another command.
-
-### Unix shells
-
-GUIs and CLIs are examples of **shells**, interfaces that sits around the
-computers innermost software and interpret the our instructions whether they
-are mouse clicks or written commands. A Unix shell is a common CLI for Unix
-systems (such as Linux and macOS) and you will often hear the term "shell" used
-colloquially to specifically refer to Unix shells. In essence, a Unix shell is
-a program which can run built-in and external programs.
-Those programs can be as complicated as climate modeling software and as simple
-as a program that creates a new directory. The simple programs which are used
-to perform stand alone tasks are usually referred to as commands.
-
-The most popular Unix shell is Bash, (the Bourne Again SHell -- named to pun
-its predecessor, the Bourne shell). Bash is the default shell on most modern
-implementations of Unix and in most packages that provide Unix-like tools for
-Windows. When Bash is first opened, you are presented with a **prompt**,
-indicating that the shell is waiting for input.
+When Bash runs it presents us with a [prompt][prompt] to indicate that it is waiting for input.
+By default,
+this prompt is a simple dollar sign:
 
 ```shell
 $
 ```
 
-Unix shells typically use `$ ` as the prompt, but may use a different symbol.
-In the examples for this lesson, we'll show the prompt as `$ `.
-Most importantly:
-when typing commands, either from these lessons or from other sources,
-*do not type the prompt*, only the commands that follow it.
+However,
+different shells may use a different symbol,
+and as we'll see later,
+we can customize the prompt to give us more information.
 
-So let's try our first command, which will tell us our username:
+> **Don't Type the Dollar Sign**
+>
+> We show the `$` prompt so that it's clear what you are supposed to type,
```

Comment:
I 100%... no a million % agree with this. We don't do this with R or Python, why do it with Bash? Plus it is a major pain in the ass to copy and paste from.... As we agreed in #220, let's remove it.

Reply:
1. I think we need to show prompts when we introduce multi-line commands (`for` loops).
2. I think we also need to show them when we have two or more users in play (which happens in the Git chapters, where the prompt includes the user ID).
3. I think we _should_ show the prompt for Python/R when we are asking people to work directly in the console rather than writing a script, so that they can more easily track how we're doing things - in this chapter, we _don't_ show the prompt when they are writing a script.
4. Looking at books by O'Reilly, and the original Kernighan et al books, they show the prompt for interactive shell sessions.

I have filed a separate discussion issue and we can re-discuss at the next meeting - please see #236.